plugins/plugins/consumer_restriction/filter.go

```diff
@@ -43,10 +43,18 @@ func (f *filter) DecodeHeaders(headers api.RequestHeaderMap, endStream bool) api
 		return f.reject("consumer not found")
 	}
 
-	_, ok := f.config.consumers[consumer.Name()]
-	if ok != f.config.allow {
-		api.LogInfof("consumerRestriction: consumer %s not allowed", consumer.Name())
-		// don't leak consumer name to the caller
+	consumerName := consumer.Name()
+	rule, consumerExists := f.config.consumers[consumerName]
+	reqMethod := headers.Method()
+	methodAllowed := rule != nil && (len(rule.Methods) == 0 || rule.Methods[reqMethod])
+
+	if f.config.allow {
```

Comment:
Look like we can simplify the comparison as `methodMatched != f.config.allow`.
If `!methodAllowed` is true then `!consumerExists` must be true.